linux/src/file_util_linux.c

```diff
@@ -0,0 +1,133 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+
+#include <string.h>
+#include <stddef.h>
+#include <stdlib.h>
+#include <fcntl.h>
+#include <inttypes.h>
+#include <stdbool.h>
+ #include <unistd.h>
+
+#include "c_pal/gballoc_hl.h" // IWYU pragma: keep
+#include "c_pal/gballoc_hl_redirect.h"
+
+#include "c_logging/logger.h"
+
+#include "c_pal/windows_defines.h"
+
+#include "c_pal/file_util.h"
+
+typedef struct CREATE_FILE_LINUX_TAG
+{
+    int h_file;
+
+} CREATE_FILE_LINUX;
+
+//Codes_SRS_FILE_UTIL_LINUX_09_002: [ file_util_create_file shall allocate memory for the file handle. file_util_create_file will succeed and return a non-NULL value. ]
+HANDLE file_util_create_file(const char* full_file_name, uint32_t access, uint32_t share_mode, LPSECURITY_ATTRIBUTES security_attributes, 
+                    uint32_t creation_disposition, uint32_t flags_and_attributes, HANDLE template_file)
+{
+    (void)flags_and_attributes;
+    
+    CREATE_FILE_LINUX* result;
+    if(((full_file_name == NULL) || (full_file_name[0] == '\0')))
+    {
+        //Codes_SRS_FILE_UTIL_LINUX_09_001: [ If the full_file_name input is either empty or NULL, file_util_create_file shall return an INVALID_HANDLE_VALUE. ]
+        LogError("Invalid arguments to file_create: const char* full_file_name=%s",
+                MU_P_OR_NULL(full_file_name));
+        result = INVALID_HANDLE_VALUE;
+    } 
+    else 
+    {
+        result = malloc(sizeof(CREATE_FILE_LINUX));
+        if(result == INVALID_HANDLE_VALUE)
```

Comment:
>INVALID_HANDLE_VALUE [](http://example.com/codeflow?start=21&length=20)

malloc returns NULL on failure. This check should be for NULL and then set the result to INVALID_HANDLE_VALUE.